lib/libvcc/vcc_xref.c

```diff
@@ -317,6 +317,29 @@ vcc_xreftable_len(struct vcc *tl, const struct symbol *sym)
 		sym_type_len = len;
 }
 
+static void v_matchproto_(symwalk_f)
+vcc_instance_info(struct vcc *tl, const struct symbol *sym)
+{
+
+	CHECK_OBJ_NOTNULL(sym, SYMBOL_MAGIC);
+	CHECK_OBJ_NOTNULL(sym->kind, KIND_MAGIC);
+	if (sym->kind != SYM_INSTANCE)
+		return;
+	AN(sym->rname);
+	Fc(tl, 0, "\t{ .p = &%s, .name = \"", sym->rname);
+	VCC_SymName(tl->fc, sym);
+	Fc(tl, 0, "\" },\n");
+}
+
+void
+VCC_InstanceInfo(struct vcc *tl)
+{
+	Fc(tl, 0, "\nconst struct vpi_ii VGC_instance_info[] = {\n");
+	VCC_WalkSymbols(tl, vcc_instance_info, SYM_NONE);
```

Comment:
Why don't we limit the scope to `SYM_INSTANCE` directly here instead of the callback? Did you try? (I haven't, yet)

Reply:
This seems to be enough:

```diff
diff --git i/lib/libvcc/vcc_xref.c w/lib/libvcc/vcc_xref.c
index 21ede56e3..89fb35cdc 100644
--- i/lib/libvcc/vcc_xref.c
+++ w/lib/libvcc/vcc_xref.c
@@ -342,9 +342,6 @@ vcc_instance_info(struct vcc *tl, const struct symbol *sym)
 {
 
        CHECK_OBJ_NOTNULL(sym, SYMBOL_MAGIC);
-       CHECK_OBJ_NOTNULL(sym->kind, KIND_MAGIC);
-       if (sym->kind != SYM_INSTANCE)
-               return;
        AN(sym->rname);
        Fc(tl, 0, "\t{ .p = &%s, .name = \"", sym->rname);
        VCC_SymName(tl->fc, sym);
@@ -355,7 +352,7 @@ void
 VCC_InstanceInfo(struct vcc *tl)
 {
        Fc(tl, 0, "\nconst struct vpi_ii VGC_instance_info[] = {\n");
-       VCC_WalkSymbols(tl, vcc_instance_info, SYM_NONE);
+       VCC_WalkSymbols(tl, vcc_instance_info, SYM_INSTANCE);
        Fc(tl, 0, "\t{ .p = NULL, .name = \"\" }\n");
        Fc(tl, 0, "};\n");
 }
```